zerver/lib/events.py

```diff
@@ -81,55 +81,61 @@ def get_raw_user_data(realm: Realm, user_profile: UserProfile, client_gravatar:
 
     if include_custom_profile_fields:
         profiles_by_user_id = get_custom_profile_field_values(realm.id)
-
-    def user_data(row: Dict[str, Any]) -> Dict[str, Any]:
-        avatar_url = get_avatar_field(
-            user_id=row['id'],
-            realm_id=realm.id,
-            email=row['delivery_email'],
-            avatar_source=row['avatar_source'],
-            avatar_version=row['avatar_version'],
-            medium=False,
-            client_gravatar=client_gravatar,
-        )
-
-        is_admin = row['role'] == UserProfile.ROLE_REALM_ADMINISTRATOR
-        is_guest = row['role'] == UserProfile.ROLE_GUEST
-        is_bot = row['is_bot']
-        # This format should align with get_cross_realm_dicts() and notify_created_user
-        result = dict(
-            email=row['email'],
-            user_id=row['id'],
-            avatar_url=avatar_url,
-            is_admin=is_admin,
-            is_guest=is_guest,
-            is_bot=is_bot,
-            full_name=row['full_name'],
-            timezone=row['timezone'],
-            is_active = row['is_active'],
-            date_joined = row['date_joined'].isoformat(),
-        )
-
-        if (realm.email_address_visibility == Realm.EMAIL_ADDRESS_VISIBILITY_ADMINS and
-                user_profile.is_realm_admin):
-            result['delivery_email'] = row['delivery_email']
-
-        if is_bot:
-            result["bot_type"] = row["bot_type"]
-            if row['email'] in settings.CROSS_REALM_BOT_EMAILS:
-                result['is_cross_realm_bot'] = True
-
-            # Note that bot_owner_id can be None with legacy data.
-            result['bot_owner_id'] = row['bot_owner_id']
-        elif include_custom_profile_fields:
-            result['profile_data'] = profiles_by_user_id.get(row['id'], {})
-        return result
-
+    else:
+        profiles_by_user_id = defaultdict(dict)
     return {
-        row['id']: user_data(row)
+        row['id']: format_user_row(realm,
+                                   user_profile = user_profile,
+                                   row=row,
+                                   client_gravatar= client_gravatar,
+                                   custom_profile_field_data = profiles_by_user_id.get(row['id'], {}),
+                                   include_custom_profile_fields=include_custom_profile_fields,
+                                   )
         for row in user_dicts
     }
 
+def format_user_row(realm: Realm, user_profile: UserProfile, row: Dict[str, Any], client_gravatar: bool,
+                    custom_profile_field_data: Optional[Dict[str, Any]] = None,
+                    include_custom_profile_fields: bool=True) -> Dict[str, Any]:
```

Comment:
`include_custom_profile_fields` is an unnecessary parameter.
If `custom_profile_field_data` is None then `include_custom_profile_fields` is False.
Otherwise, `include_custom_profile_fields` is True.
Can you remove `include_custom_profile_fields`?

Reply:
`get_raw_user_data`  uses this to initialize custom_profile_data in case it is set to true . 
Removing this would mean that get_raw_user_data will always return dictionaries with  `'profile_data'` : `custom_profile_user_data` .

Reply:
I'm saying that it shouldn't be passed as a parameter to `format_user_row` (here on line 99) when we're already passing the `custom_profile_field_data` parameter as a Optional[Dict[str, Any]] type. `get_raw_user_data` can and should continue to use it, but it doesn't need to be sent to `format_user_row` because of the logic that I stated in my original comment. In other words, line 135 in `events.py` can just be
```python3
elif custom_profile_field_data is not None:
````
instead of relying on `include_custom_profile_fields`.

Reply:
That is causing some tests to fail , I'll try to fix and re-push .